lib/config/config-file.js

```diff
@@ -210,6 +244,8 @@ function loadConfigFile(filePath) {
             default:
                 config = loadLegacyConfigFile(filePath);
         }
+    } else if (filePath.indexOf("eslint-plugin") >= 0) {
```

Comment:
`whatever-eslint-plugin` isn't going to pass checks in `resolve`. Resolve's job is to normalize name of the config/plugin to the standard format. If it returned something that has `eslint-plugin` in it, that means that it used to have a `plugin:` prefix.